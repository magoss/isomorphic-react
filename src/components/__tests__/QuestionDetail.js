import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail';
import renderer from 'react-test-renderer';
import React from 'react'


describe('Test Question Details Component', () => {
  describe('The container element', () => {
    describe('mapStateToProps', () => {
      it('should map the state to props correctly', () => {
        const appState = { questions: [{
          question_id: 43,
          body: 'some body 43'
        }, {
          question_id: 42,
          body: 'some body 42'
        }] };
        const ownProps = {
          question_id: 42
        };
        const componentState = mapStateToProps(appState, ownProps);

        expect(componentState).toEqual({ question_id: 42, body: 'some body 42' });
      });
    });
  });

  describe('The display element', () => {
    it('should not regress', () => {
      const tree = renderer.create(
        <QuestionDetailDisplay 
          title="Some title changed"
          body="Some body"
          answer_count={2}
          tags={['tag1', 'tag2']}
        />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
    
  });
});