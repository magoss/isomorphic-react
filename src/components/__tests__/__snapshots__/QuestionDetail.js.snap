// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Question Details Component The display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Some title changed
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          tag1
        </code>
        <code>
          tag2
        </code>
      </div>
    </div>
    <div>
      <p>
        Some body
      </p>
    </div>
    <div>
      2
       Answers
    </div>
  </div>
</div>
`;
